core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java

```diff
@@ -61,19 +65,71 @@ public String getSignalingType() {
         }
     }
 
+    private DiTrackEdge getLinkedTrackSection(DiTrackInfra infra, DiTrackEdge curEdge) {
+        var graph = infra.getDiTrackGraph();
+        var nodes = graph.incidentNodes(curEdge);
+        var endNode = nodes.nodeV();
+        var neighbors = graph.outEdges(endNode);
+        // look for a neighboring track section
+        // neighboring edges are either switch branches or track sections
+        DiTrackEdge neighborTrack = null;
+        for (var neighbor : neighbors) {
+            if (neighbor.getEdge() instanceof TrackSection) {
+                neighborTrack = neighbor;
+                break;
+            }
+        }
+        return neighborTrack;
+    }
+
+    private DiDetector getNextDetector(DiTrackInfra infra, DiTrackEdge curEdge) {
+        // if that fails, look for linked track sections
+        // var curEdge = infra.getEdge(trackSection, Direction.FORWARD);
```

Comment:
That comment should probably be near the call site (what is "that"?)

Also, commented code